File: app/schemas/stats.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

Health = Literal["healthy", "degraded", "down"]


class StatsHealth(BaseModel):
    model_config = ConfigDict(ser_json_by_alias=True, populate_by_name=True)

    embedding: Health
    llm: Health
    vector_db: Health = Field(serialization_alias="vectorDb")


class StatsOverviewOut(BaseModel):
    model_config = ConfigDict(ser_json_by_alias=True, populate_by_name=True)

    total_documents: int = Field(serialization_alias="totalDocuments")
    indexed_chunks: int = Field(serialization_alias="indexedChunks")
    active_collections: int = Field(serialization_alias="activeCollections")
    embedding_model: str = Field(serialization_alias="embeddingModel")
    llm_provider: str = Field(serialization_alias="llmProvider")
    avg_query_latency_ms: int = Field(serialization_alias="avgQueryLatencyMs")
    storage_usage_bytes: int = Field(serialization_alias="storageUsageBytes")
    queue_depth: int = Field(serialization_alias="queueDepth")
    active_jobs: int = Field(serialization_alias="activeJobs")
    health: StatsHealth


class ActivityPointOut(BaseModel):
    model_config = ConfigDict(ser_json_by_alias=True, populate_by_name=True)

    date: str
    queries: int
    uploads: int
    errors: int

File: frontend/lib/mock/stats.ts
import type { ActivityPoint, StatsOverview } from "@/lib/types";
import { listMockDocuments } from "@/lib/mock/documents";
import { MOCK_COLLECTIONS } from "@/lib/mock/collections";

export function getMockStats(): StatsOverview {
  const docs = listMockDocuments();
  const chunks = docs.reduce((s, d) => s + d.chunkCount, 0);
  return {
    totalDocuments: docs.length,
    indexedChunks: chunks,
    activeCollections: MOCK_COLLECTIONS.length,
    embeddingModel: "text-embedding-3-large",
    llmProvider: "gpt-4.1-mini",
    avgQueryLatencyMs: 840,
    storageUsageBytes: 2_400_000_000 + docs.length * 50_000,
    queueDepth: 2,
    activeJobs: 3,
    health: {
      embedding: "healthy",
      llm: "healthy",
      vectorDb: "degraded",
    },
  };
}

export function getMockActivity(): ActivityPoint[] {
  const days = 7;
  return Array.from({ length: days }, (_, i) => {
    const d = new Date();
    d.setDate(d.getDate() - (days - 1 - i));
    return {
      date: d.toISOString().slice(0, 10),
      queries: 120 + i * 18 + ((i * 7) % 23),
      uploads: 4 + (i % 4),
      errors: i === 3 ? 2 : 0,
    };
  });
}
